src/main/java/com/epam/rd/autotasks/springemployeecatalog/repository/EmployeeRepository.java

```diff
@@ -0,0 +1,117 @@
+package com.epam.rd.autotasks.springemployeecatalog.repository;
+
+import com.epam.rd.autotasks.springemployeecatalog.domain.Employee;
+import com.epam.rd.autotasks.springemployeecatalog.extractors.ExtractorFactory;
+import com.epam.rd.autotasks.springemployeecatalog.extractors.ExtractorFactoryImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.ALL_EMPLOYEES;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.BY_DEPARTMENT_ID;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.BY_DEPARTMENT_NAME;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.BY_ID;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.BY_MANAGER;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.DIGIT_REGEX;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.EMPLOYEE_BY_ID_SELECT;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.HIREDATE_CASE;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.LASTNAME_CASE;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.LIMIT_OFFSET;
+import static com.epam.rd.autotasks.springemployeecatalog.constants.AppConstants.POSITION_CASE;
+import static com.epam.rd.autotasks.springemployeecatalog.utils.StatementUtils.getPreparedStatementCreator;
+
+@Repository
+public class EmployeeRepository {
+    public static final String ORDER_BY_LASTNAME = " ORDER BY LASTNAME_e";
+    public static final String ORDER_BY_HIREDATE = " ORDER BY HIREDATE_e";
+    public static final String ORDER_BY_POSITION = " ORDER BY POSITION_e";
+    public static final String ORDER_BY_SALARY = " ORDER BY SALARY_e";
+    private final JdbcTemplate jdbcTemplate;
+    private final ExtractorFactory factory;
+
+    @Autowired
+    public EmployeeRepository(JdbcTemplate jdbcTemplate, ExtractorFactoryImpl factory) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.factory = factory;
+    }
+
+    public Employee getEmployeeById(Long id, boolean withChain) {
+
+        String finalQuery;
+        if (withChain) {
+            finalQuery = String.format(EMPLOYEE_BY_ID_SELECT, id);
+        } else {
+            finalQuery = String.format(ALL_EMPLOYEES + BY_ID, id);
+        }
+
+        return Objects.requireNonNull(jdbcTemplate.query(
+                        getPreparedStatementCreator(finalQuery),
+                        factory.getExtractor(withChain)))
+                .get(0);
```

Comment:
Objects.requireNonNull is not protect you from empty list. NPE is possible here.